lib/remote/httpserverconnection.hpp

```diff
@@ -45,6 +44,15 @@ class HttpServerConnection final : public Object
 
 	HttpServerConnection(const String& identity, bool authenticated, const Shared<AsioTlsStream>::Ptr& stream, boost::asio::io_context& io);
 
+	/**
+	 * Tries to asynchronously shut down the SSL stream and underlying socket.
+	 *
+	 * It is important to note that this method should only be called from within a coroutine that uses `m_IoStrand`.
+	 *
+	 * @param yc boost::asio::yield_context The coroutine yield context which you are calling this method from.
+	 */
```

Comment:
Are both of you waiting for a reaction from the other one now? ;)

> Aren't such above the function **definitions** and not the declarations like here?

On that topic: yes, that seems to often be the case in the Icinga 2 code. (Though I don't really understand why actually, so if you know, please enlighten me.)

Reply:
"often" like _there's a minority of locations like Yonas' one_ or like _there's a minority(?) of locations w/o code doc at all_?

Reply:
"often" like I did not do any exact statistics. I don't know how many exceptions there are or if there are exceptions to that at all.

Reply:
In contrast to pure virtual methods this one even _has_ a definition where the comment can be moved to.

> I don't really understand why actually, so if you know, please enlighten me.

I can only say _why I want_ this:

Sources are already large. A few comments more don't matter.
Headers, in contrast, are already kind of a compact list of available methods, a cheat sheet.

Reply:
This is simply your personal _subjective_ viewpoint. [As Johannes](https://icinga.com/blog/2024/10/30/code-reviews-how-do-they-work/) pointed out, if you don't have objective reasons to back up your argument, you should better not start it.

I, on the other hand, believe that the documents should be in the header files, because if you want to know what a method does, you simply go to the header files and browse its documentation without having to search for the actual implementation. This is almost always the case with libraries, because you as a user should not have access to the actual source files, but only to the headers that you have included. So, please stop delaying this PR unnecessarily and enforcing to dismiss an already approved PR for nothing.

Reply:
> browse its documentation without having to **search** for the actual implementation

In CLion, there's a pair of arrows next to each method declaration allowing you to navigate to the definition quickly if you're interested in the docs.

Reply:
I'm going to change this now, not because I'm convinced of your personal opinion, but because I don't want to waste my time discussing a pointless topic that you shouldn't have brought up in the first place.